File: 169. Majority Element/Approach03.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int result = nums[0];
        int cnt = 1;
        for(int i=1; i<n; i++){
            if(cnt==0){
                result = nums[i];
            }
            
            if(nums[i]==result){
                cnt++;
            }
            else if(nums[i]!=result){
                cnt--;
            }
        }
        return result;
    }
};

File: 169. Majority Element/Approach03.py
from typing import *

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        result = nums[0]
        cnt = 1

        for i in range(1,n):
            if(cnt==0):
                result = nums[i]
            if(nums[i]==result):
                cnt += 1
            elif(nums[i] != result):
                cnt -= 1
        return result